pallets/dactr/src/lib.rs

```diff
@@ -520,6 +641,193 @@ pub mod pallet {
 
 			Ok(().into())
 		}
+
+		#[pallet::call_index(9)]
+		#[pallet::weight(T::WeightInfo::register_blob_offence())]
+		pub fn register_blob_offence(
+			origin: OriginFor<T>,
+			offence_key: OffenceKey,
+			voucher: ValidatorVoucher,
+		) -> DispatchResultWithPostInfo {
+			ensure_none(origin)?;
+
+			// Ensure the key is valid based on the kind and params
+			ensure!(offence_key.is_valid(), Error::<T>::InvalidOffenceKey);
+
+			// Get validators list
+			let validators = T::SessionDataProvider::validators();
+
+			// Check given session
+			let provided_session = voucher.session_index;
+			let current_session = T::ValidatorSet::session_index();
+			ensure!(
+				provided_session == current_session,
+				Error::<T>::InvalidVoucherSession
+			);
+
+			// Checks only run outside of tests and runtime benchmarks
+			let Some(validator_id) =
+				T::AccountId::decode(&mut &voucher.validator.encode()[..]).ok()
+			else {
+				return Err(Error::<T>::InvalidVoucherValidator.into());
+			};
+
+			#[cfg(not(any(test, feature = "runtime-benchmarks")))]
```

Comment:
To make it more crystal clear, make it a function and then do an early return if it is a test or a runtime bench